packages/integration-sdk-core/src/types/config.ts

```diff
@@ -83,10 +83,18 @@ export interface InvocationConfig<
   /**
    * Optionally define the number of steps to execute in parallel.
    *
-   * If this value is not provided the concurrency defaults to
+   * If this value is not provided, the concurrency defaults to
    * Number.POSITIVE_INFINITY
    */
   stepConcurrency?: number;
+  /**
+   * Optionally collect the encountered keys and return them in
+   * the integration results. This can result in a large quantity
+   * of keys being returned.
+   *
+   * If this value is not provided, the default is false.
+   */
+  collectEncounteredKeys?: boolean;
```

Comment:
How is this set? Do we have to re-deploy the integration when we want to enable this?

Reply:
This would be set in the invocationConfig which can be loaded dynamically with the loadInvocationConfig hook. We can toggle this at runtime this way.